File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;
class Solution {
    public int solution(int[] elements) {
        Set<Integer> set = new HashSet<>();
        for(int i = 1; i < elements.length + 1; i++){
            for(int j = 0; j < elements.length; j++){
                int tmp = 0;
                for(int k = j; k < j + i; k++){
                    tmp += elements[k % elements.length];
                }
                set.add(tmp);
            }
        }
        return set.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    var answer = new Set();
    for(let size = 1; size <= elements.length; size++){
        for(let start = 0; start < elements.length; start++){
            let tmp = 0;
            for(let idx = start; idx < start + size; idx++){
                tmp += elements[idx % elements.length];
            }
            answer.add(tmp);
        }
    }
    return answer.size;
}